addons/crm_livechat/tests/test_discuss_channel_access.py

```diff
@@ -35,3 +36,28 @@ def test_access_channel_from_lead(self):
                 f"user_grp={user_grp}, has_lead={has_lead}, expected_result={expected_result}",
             )
 
+    def test_cannot_link_lead_to_restricted_channel(self):
+        user = new_test_user(self.env, login="bob_user", groups="sales_team.group_sale_salesman")
+        channel = (
+            self.env["discuss.channel"]
+            .create(
+                {
+                    "name": f"Visitor #11",
+                    "livechat_operator_id": self.env.user.partner_id.id,
+                    "channel_type": "livechat",
+                }
+            )
+            .with_user(user)
+        )
+        self.assertFalse(channel.has_access("read"))
+        with self.assertRaises(
+            AccessError, msg="You cannot link a lead to a channel you don't have access to."
+        ):
+            self.env["crm.lead"].with_user(user).create({"name": "lead", "livechat_channel_id": channel.id})
+
+        lead = self.env["crm.lead"].with_user(user).create({"name": "lead"})
+        with self.assertRaises(
+            AccessError, msg="You cannot link a lead to a channel you don't have access to."
```

Comment:
Might be useful to have 2 different error messages to help seeing which is crashing (create/write).